src/components/AddItem.jsx

```diff
@@ -2,58 +2,65 @@ import { useState } from "react";
 const AddItem = ({ onNewItem }) => {
   const [itemName, setItemName] = useState();
   const [itemDate, setItemDate] = useState();
+  const [  errorMsg , setErrorMsg ] = useState();
 
-  const handleNameChange = (event) => {
-    setItemName(event.target.value);                                                    //  this function handle the state change of itemName
-  };
-  const handleDateChange = (event) => {
-    setItemDate(event.target.value);                                                    //  this function handle the state change of itemDate
-  };
-  const handleAddButton = () => {
+  const today = ()=>
+  {
     const currentDate = new Date();                                                     //  this function check the inserted date must not be less then todays date
-    var dd = currentDate.getDate();
-    var mm = currentDate.getMonth() + 1;
-    var yyyy = currentDate.getFullYear();
+    let dd = currentDate.getDate();
+    let mm = currentDate.getMonth() + 1;
+    let yyyy = currentDate.getFullYear();
     if (dd < 10) {
       dd = "0" + dd;
     }
     if (mm < 10) {
       mm = "0" + mm;
     }
-    var today = yyyy + "-" + mm + "-" + dd;
+    let today = yyyy + "-" + mm + "-" + dd;
+    return today;
+  }
+  
+  const todayDate = today();
+
+  const handleAddButton = () => {
+    
 
     itemName && itemDate                                                              //  ternarry opprator is used to ensure both the fields are populated
-      ? itemDate >= today
-        ? onNewItem(itemName, itemDate)
-        : console.log("GIVE VALID DATE")
-      : console.log("GIVE VALID INPUT");
-    setItemName("");
-    setItemDate("");
+      ? itemDate >= todayDate
+        ? (onNewItem(itemName, itemDate), setItemName(""), setItemDate(""),setErrorMsg("")  )
```

Comment:
`setErrorMsg` can be set initially at the time of declaration. Similarly for `setItemDate`.